lms/djangoapps/course_api/blocks/tests/test_api.py

```diff
@@ -226,8 +226,8 @@ def test_query_counts_cached(self, store_type, with_storage_backing):
             )
 
     @ddt.data(
-        (ModuleStoreEnum.Type.split, 2, True, 23),
-        (ModuleStoreEnum.Type.split, 2, False, 13),
+        (ModuleStoreEnum.Type.split, 2, True, 24),
+        (ModuleStoreEnum.Type.split, 2, False, 14),
```

Comment:
Note to reviewer: I'm just making tests pass with these changes -- it might not be the definitive implementation. Still, all these increases are making me second-guess the flag implementation.

Reply:
Just so you know, the tests are failing because of the query count, so I'll leave them be until we have a review and I hear what you think about the extra queries.

Reply:
I think that fall largely out of your control in this PR. The waffle flag require a query here but it more than makes up in the flexibility in the rollout.